Add assertions and comment on the utility class in Spring Integration IP.

ZahidFKhan opened the pull request:
Added assertion on the utility class, so that it can't be mistakenly ever initialized in the class.

<!--
Thanks for contributing to Spring Integration. 
Please provide a brief description of your pull-request and reference any related issue numbers (prefix references with #) or StackOverflow questions.

See the [Contributor Guidelines for more information](https://github.com/spring-projects/spring-integration/blob/main/CONTRIBUTING.adoc).
-->

Commit:
Add assertions and comment on the utility class.
Added assertion on the utility class, so that it can't be mistakenly ever initialized in the class.

artembilan commented on spring-integration-ip/src/main/java/org/springframework/integration/ip/util/TestingUtilities.java:
```diff
@@ -33,7 +33,9 @@
 
+	// Non-instantiable utility class
 	private TestingUtilities() {
+		throw new AssertionError();
```
This is `private`, so it cannot be instantiate from the outside of the class.
Even if you do some hardcore via reflection, it still does not harm since there is no any state in these classes to break something.
I'm not sure that is going to be mistake in the framework code to call this ctor directly in this class: we always review changes, so such a code won't pass it to the mainstream.

Please, elaborate on the purpose of this change.

Thanks

ZFK07 replied:
I agree @artembilan that this is a `private` utility class and it can't be instantiated from outside of the class.

The `AssertionError` isn’t strictly required at all but it provides insurance in case the constructor is accidentally invoked from within the class not now but at some point in time. It guarantees that the class will never be instantiated under any circumstances. It is therefore wise to include a comment as well.

artembilan replied:
OK. So, add, please, a message to that error and revert that JavaDocs format. We prefer to not have blank line in the method JavaDocs.
Plus, bring that blank line before `@since` on the class level - for better readability 😄

artembilan replied:
Tell me, please, if you are going to address other my suggestions.
Thanks

ZFK07 replied:
Yes, @artembilan I am working on the above suggestions. :smile: 
And I haven't modified any of the java docs. Seems there is some problem with CRLF and LF in my IDE. :see_no_evil:

ZahidFKhan committed:
Update RegexUtils.java

Undo the removal of extra space.

ZahidFKhan committed:
Update RegexUtils.java

Removed extra space.

ZahidFKhan committed:
Add exception message for RegexUtils

ZahidFKhan committed:
Add Exception Message for TestingUtilities.java

Add exception message "Class Instantiation not allowed." for the TestingUtilities

ZahidFKhan committed:
Update TestingUtilities.java

ZFK07 commented:
Hi @artembilan would you like to verify again, please?

ZFK07 commented:
Thank you so much @artembilan 

This is my first contribution to the Open Source, definitely, I would love to contribute more to the Spring Ecosystem.
Looking forward to working with you in the future. 😄😇
